src/shared/apis/queryCache.ts

```diff
@@ -155,11 +191,10 @@ const hasWindowVars = (variables: VariableAssignment[]): boolean =>
 export const getCachedResultsOrRunQuery = (
   orgID: string,
   query: string,
-  state: AppState
+  usedVars: Variable[]
 ): CancelBox<RunQueryResult> => {
   const queryID = `${hashCode(query)}`
   event('Starting Query Cache Process ', {context: 'queryCache', queryID})
-  const usedVars = filterUnusedVarsBasedOnQuery(getAllVariables(state), [query])
```

Comment:
made this a parent concern to make the function more isolated

Reply:
the fetching of the variables should totally be a parent concern, but i think there's value in having an interface where you can throw it some dirty values and it'll clean them up for you